backend/spec/features/admin/orders/order_details_spec.rb

```diff
@@ -399,6 +399,31 @@
               expect(page).not_to have_selector('.fa-arrows-h')
               expect(page).not_to have_selector('.fa-trash')
             end
+
+            context 'and on the cart page the cart is emptied' do
+              it 'should show the empty cart page' do
+                order = create(:order_ready_to_ship)
+
+                visit spree.cart_admin_order_path(order)
+                order.fulfill!
+
+                ## simulate shipping order in another tab
+                shipment = order.shipments.first
+                order.shipping.ship(
+                  inventory_units: shipment.inventory_units,
+                  stock_location: shipment.stock_location,
+                  address: order.ship_address,
+                  shipping_method: shipment.shipping_method
+                )
+
+                accept_alert 'Are you sure you want to delete this record?' do
+                  click_on 'Empty Cart'
+                end
+
+                expect(page.current_path).to eq(spree.cart_admin_order_path(order))
+                expect(page).not_to have_content(order.line_items.first.variant.sku)
```

Comment:
This can lead to false positives. Can we expect an empty order message instead?

```suggestion
                expect(page).to have_content I18n.t('spree.your_order_is_empty_add_product')
```

Reply:
Thanks @tvdeyen. I ended up re-writting most of this spec in the latest revision, so let me know what you think :)